AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m

```diff
@@ -1,33 +1,162 @@
+  NSDictionary *dic = [MSCosmosDb defaultHeaderWithPartition:kMSPartitionTest dbToken:kMSTokenTest additionalHeaders:nil];
+
+  // Then
+  XCTAssertNotNil(dic);
+  XCTAssertTrue(dic[@"Content-Type"]);
+}
+
+- (void)testDocumentUrlWithTokenResultWithStringToken {
+
+  // If
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithString:kMSTokenTest];
+
+  // When
+  NSString *result = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:kMSDocumentIdTest];
+
+  // Then
+  XCTAssertNotNil(result);
+}
+
+- (void)testDocumentUrlWithTokenResultWithObjectToken {
+
+  // When
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:[self prepareMutableDictionary]];
+  NSString *testResult = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:@"documentId"];
+
+  // Then
+  XCTAssertNotNil(testResult);
+  XCTAssertTrue([testResult containsString:kMSDocumentIdTest]);
+  XCTAssertTrue([testResult containsString:kMSDbAccountTest]);
+  XCTAssertTrue([testResult containsString:kMSDbNameTest]);
+  XCTAssertTrue([testResult containsString:kMSDbCollectionNameTest]);
+}
+
+- (void)testDocumentUrlWithTokenResultWithDictionaryToken {
+
+  // If
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:[self prepareMutableDictionary]];
+
+  // When
+  NSString *testResult = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:kMSDocumentIdTest];
+
+  // Then
+  XCTAssertNotNil(testResult);
+  XCTAssertTrue([testResult containsString:kMSDocumentIdTest]);
+  XCTAssertTrue([testResult containsString:kMSDbAccountTest]);
+  XCTAssertTrue([testResult containsString:kMSDbNameTest]);
+  XCTAssertTrue([testResult containsString:kMSDbCollectionNameTest]);
+}
+
+- (void)testPerformCosmosDbAsyncOperationWithHttpClientWithAdditionalParams {
+
+  // If
+  MSCosmosDbIngestion *httpClient = [[MSCosmosDbIngestion alloc] init];
+  MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:[self prepareMutableDictionary]];
+  NSMutableDictionary *additionalHeaders = [NSMutableDictionary new];
+  additionalHeaders[@"Foo"] = @"Bar";
+  __block BOOL completionHandlerCalled = NO;
+  XCTestExpectation *completeExpectation = [self expectationWithDescription:@"Task finished"];
+  MSCosmosDbCompletionHandler handler = ^(NSData *_Nullable data, NSError *_Nullable error) {
+    completionHandlerCalled = YES;
+    [completeExpectation fulfill];
+  };
+
+  // When
+  [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:httpClient
+                                              tokenResult:tokenResult
+                                               documentId:kMSDocumentIdTest
+                                               httpMethod:kMSHttpMethodGet
+                                                     body:nil
+                                        additionalHeaders:additionalHeaders
+                                        completionHandler:handler];
+  [self waitForExpectationsWithTimeout:5 handler:nil];
```

Comment:
Reason of using 5 sec for timeout?

Reply:
Just the gap to let system send request and get response; Depending on system, it takes about 0.3 sec for the test to be finished. But well, yes, reducing the timeout might be reasonable

Reply:
Are we sending actual requests here? I was sure that we have only unit tests and need to mock all actual requests.

Sending actual requests slows down tests execution and makes them unstable because test result depends on other services.